pageserver/src/tenant/size.rs

```diff
@@ -183,7 +183,13 @@ pub(super) async fn gather_inputs(
         // new gc run, which we have no control over. however differently from `Timeline::gc`
         // we don't consider the `Timeline::disk_consistent_lsn` at all, because we are not
         // actually removing files.
-        let mut next_gc_cutoff = cmp::min(gc_info.horizon_cutoff, gc_info.pitr_cutoff);
+        //
+        // We only consider [`GcInfo::pitr_cutoff`], and not [`GcInfo::horizon_cutoff`], because from
+        // a user's perspective they have only requested retention up to the time bound (pitr_cutoff), rather
+        // than a space bound (horizon cutoff).  This means that if someone drops a database and waits for their
+        // PITR interval, they will see synthetic size decrease, even if we are still storing data inside
+        // horizon_cutoff.
+        let mut next_gc_cutoff = gc_info.pitr_cutoff;
```

Comment:
Weird that none of the tests failed. Creating a test which reproduces the problem seems quite straightforward. I'd run the disabled ones as well if any of them already are sensitive.

Reply:
Luckily even the ignored tests in `test_runner/regress/test_tenant_size.py` work on pg14 before this change.

Reply:
Now that we have a solution for reproducable initdb_lsn one could reconsider adding tighter values there as well, but of course there is the caveat of ignored but still increasing `last_record_lsn` cases which are probably a big chunk of the instability already.

Reply:
I finally got back around to this.  Good flag that there were tests that really ought to have failed.

I hadn't realized the synthetic size tests were in such a degraded state: opened https://github.com/neondatabase/neon/issues/7138 to track that.

`test_single_branch_get_tenant_size_grows` has a case for dropping a table, which checks that size _increases_, but it didn't cover this change because the test was running with a zero PITR interval.

Reinstating and extending that test to use a 1ms PITR interval to exercise this PR, I found that `update_gc_info` has a bug (I think).  If the PITR time point is more recent than the most recent commit, then it falls back to using the size-based GC horizon -- that doesn't make sense: if the PITR time point is more recent than the recent commit, it should be interpreted that the PITR LSN is the head of the log.  I've added a commit that changed that behavior.

Reply:
Yes, we added those extra tests back then but didn't leave them running because other tests should cover the cases. Those at least made it more straightforward back then to consider different situations.

> If the PITR time point is more recent than the most recent commit, then it falls back to using the size-based GC horizon -- that doesn't make sense: if the PITR time point is more recent than the recent commit, it should be interpreted that the PITR LSN is the head of the log. I've added a commit that changed that behavior.

I cannot imagine where the two would not be combined by `min`, before this PR that is.